Return null if accessUrl generation times out BT-259

kshakir opened the pull request.

kshakir committed:
Return null if accessUrl generation times out

kshakir commented on martha/martha_v3.js:
```diff
@@ -527,7 +532,26 @@ async function retrieveFromServers(params) {
+            upstream client to martha_v3 might get a response if there's time left.
+             */
+            // noinspection JSCheckFunctionSignatures
+            const accessUrlAttempt = await Promise.race([
+                apiAdapter.getJsonFrom(httpsAccessUrl, accessTokenAuth),
+                sleep(REQUEST_TIMEOUT_MILLIS, REQUEST_TIMEOUT_RESULT),
+            ]);
```
To do: if there's _any_ error returned, even not a timeout, then keep going and return `accessUrl: null`

breilly2 replied:
Agreed. Once we have that, I believe this is good to go.

kshakir replied:
Updated and added a test.

breilly2 commented on common/helpers.js:
```diff
@@ -1,6 +1,6 @@
 const url = require('url');
 const config = require('../common/config');
-
+const sleep = require('util').promisify(setTimeout);
```
I believe this will be fine. https://nodejs.org/api/util.html#util_util_promisify_original mentions the pitfall with `this`, but that isn't a problem here.

kshakir committed:
Merge branch 'dev' into ks_access_url_timeout_null

kshakir committed:
Multiple updates:
- Continue when errors occur trying to retrieve an access URL
- Switched tests to use gmkf instead of bdc due to changes in dev branch

kshakir commented:
This is going to be redone. Closing so this doesn't confuse the Jira board but feel free to re-use the testing logic.
